src/app/server/Dnssd.cpp

```diff
@@ -234,11 +234,21 @@ CHIP_ERROR DnssdServer::Advertise(bool commissionableNode, chip::Dnssd::Commissi
         advertiseParameters.SetDeviceType(chip::Optional<uint32_t>::Value(val32));
     }
 
-    char deviceName[chip::Dnssd::kKeyDeviceNameMaxLength + 1];
-    if (DeviceLayer::ConfigurationMgr().IsCommissionableDeviceNameEnabled() &&
-        DeviceLayer::ConfigurationMgr().GetCommissionableDeviceName(deviceName, sizeof(deviceName)) == CHIP_NO_ERROR)
+    char deviceName[chip::Dnssd::kKeyDeviceNameMaxLength];
+    MutableCharSpan deviceNameSpan(deviceName);
+    if (DeviceLayer::ConfigurationMgr().IsCommissionableDeviceNameEnabled())
     {
-        advertiseParameters.SetDeviceName(chip::Optional<const char *>::Value(deviceName));
+        CHIP_ERROR err;
+        if ((err = DeviceLayer::ConfigurationMgr().GetCommissionableDeviceName(deviceNameSpan)) == CHIP_NO_ERROR)
+        {
+            // Appending a null terminator since SetDeviceName expects a null terminated string
+            deviceName[deviceNameSpan.size()] = '\0';
```

Comment:
But ... if the span in fact filled the whole buffer, this is writing out of bounds, no?

This needs to be something like:

```
// Leave room for a null terminator.
char deviceName[Dnssd::kKeyDeviceNameMaxLength + 1];
MutableCharSpan deviceNameSpan(deviceName, Dnssd::kKeyDeviceNameMaxLength);
```
so that we make sure we have space for that null.

Reply:
this is true.